x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/template/TemplateUtils.java

```diff
@@ -91,13 +99,22 @@ public static void validate(BytesReference source) {
         }
     }
 
+    private static BytesReference replaceVariables(BytesReference input, String version, String versionProperty,
+                                                         Map<String, String> variables) {
+        String template = replaceVariable(input.utf8ToString(), versionProperty, version);
+        for (Map.Entry<String, String> variable : variables.entrySet()) {
+            template = replaceVariable(template, variable.getKey(), variable.getValue());
+        }
+        return new BytesArray(template);
+    }
+
     /**
-     * Filters the source: replaces any template version property with the version number
+     * Replaces all occurences of given variable with the value
      */
-    public static String filter(BytesReference source, String version, String versionProperty) {
-        return Pattern.compile(versionProperty)
-                .matcher(source.utf8ToString())
-                .replaceAll(version);
+    public static String replaceVariable(String input, String variable, String value) {
+        return Pattern.compile(variable)
```

Comment:
Because the only usage of this function is in `TemplateUtils`, maybe we should add the [`LITERAL`](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html#LITERAL) flag here rather than quoting higher in the call stack? I think we're unlikely to need full regex support here.

Reply:
Done